crates/project-model/src/cargo_workspace.rs

```diff
@@ -288,7 +288,22 @@ impl CargoWorkspace {
         locked: bool,
         progress: &dyn Fn(String),
     ) -> anyhow::Result<(cargo_metadata::Metadata, Option<anyhow::Error>)> {
-        Self::fetch_metadata_(cargo_toml, current_dir, config, sysroot, locked, false, progress)
+        let res = Self::fetch_metadata_(
+            cargo_toml,
+            current_dir,
+            config,
+            sysroot,
+            locked,
+            false,
+            progress,
+        );
+        if let Ok((_, Some(ref e))) = res {
+            tracing::warn!(
+                "`cargo metadata` failed on `{cargo_toml}`, but retry with `--no-deps` succeeded"
+            );
+            tracing::debug!("{e:?}");
```

Comment:
```suggestion
            tracing::warn!(
                %cargo_toml,
                %e,
                "`cargo metadata` failed on `{}`, but retry with `--no-deps` succeeded"
            );
```

Reply:
That looks better, but doesn't print the `stderr` though, while `tracing::debug!("{e:?}");` does. I'd really like to see a short warning with a default log level and a detailed error message explaining the cause at debug/trace level. Not sure that the best way is to achieve that.

Reply:
this should include the command's `stderr`:

```rust
tracing::warn!(
    %cargo_toml,
    ?e,
    "`cargo metadata` failed, but retry with `--no-deps` succeeded"
);
```

personally, I think including `stderr` is worth the noise, but.

Reply:
Ah yes, that makes sense, I can change to `?e` . @Veykril what do you think?

Reply:
Thanks both! I applied @davidbarsky's suggestion and replaced `%e` with `?e` .